src/native/overlay/list.rs

```diff
@@ -0,0 +1,168 @@
+            match event {
+                Event::Mouse(mouse::Event::CursorMoved { .. }) => {
+                    *self.hovered_option = Some(
+                        ((cursor_position.y - bounds.y)
+                            / f32::from(self.style.text_size + (self.style.padding * 2)))
+                            as usize,
+                    );
+                }
+                Event::Mouse(mouse::Event::ButtonPressed(mouse::Button::Left))
+                | Event::Touch(touch::Event::FingerPressed { .. }) => {
+                    *self.hovered_option = Some(
+                        ((cursor_position.y - bounds.y)
+                            / f32::from(self.style.text_size + (self.style.padding * 2)))
+                            as usize,
+                    );
+
+                    if let Some(index) = self.hovered_option {
+                        if let Some(option) = self.options.get(*index) {
+                            *self.last_selection = Some(option.clone());
+                        }
+                    }
+
+                    status = self
+                        .last_selection
+                        .take()
+                        .map_or(event::Status::Ignored, |last| {
+                            messages.push((self.on_selected)(last));
+                            event::Status::Captured
+                        });
+                }
+                _ => {}
+            }
+        }
+
+        status
+    }
+
+    fn draw(
+        &self,
+        renderer: &mut Renderer,
+        _defaults: &Renderer::Defaults,
+        layout: Layout<'_>,
+        cursor_position: Point,
+        viewport: &Rectangle,
+    ) -> Renderer::Output {
+        self::Renderer::draw(
+            renderer,
+            layout.bounds(),
+            cursor_position,
+            viewport,
+            self.options,
+            *self.hovered_option,
+            self.font,
+            &self.style,
+        )
+    }
+}
+
+/// The renderer of a [`List`].
+///
+/// Your [renderer] will need to implement this trait before being
+/// able to use a [`List`] in your user interface.
+///
+/// [renderer]: crate::renderer
+pub trait Renderer: scrollable::Renderer + container::Renderer + text::Renderer {
+    /// Draws the list of options of a [`List`].
+    #[allow(clippy::too_many_arguments)]
+    fn draw<T: ToString>(
+        &mut self,
+        bounds: Rectangle,
+        cursor_position: Point,
+        viewport: &Rectangle,
+        options: &[T],
+        hovered_option: Option<usize>,
+        font: Self::Font,
+        style: &selection_list::Style,
+    ) -> Self::Output;
+}
+
+impl<'a, T, Message, Renderer> From<List<'a, T, Message, Renderer>>
+    for Element<'a, Message, Renderer>
+where
+    T: ToString + Clone,
+    Message: 'a,
+    Renderer: 'a + self::Renderer,
+{
+    fn from(list: List<'a, T, Message, Renderer>) -> Element<'a, Message, Renderer> {
+        Element::new(list)
+    }
+}
```

Comment:
This file could now be placed in a submodule of the selection list instead of the overlay module, as it is no overlay anymore.

Reply:
I will move it to its own module